spark-jobs/src/main/scala/filodb.downsampler/BatchDownsampler.scala

```diff
@@ -124,19 +107,21 @@ object BatchDownsampler extends StrictLogging with Instance {
     }
     val rawPartsToFree = ArrayBuffer[PagedReadablePartition]()
     val downsampledPartsPartsToFree = ArrayBuffer[TimeSeriesPartition]()
-    rawPartsBatch.foreach { rawPart =>
-      downsamplePart(rawPart, rawPartsToFree, downsampledPartsPartsToFree,
-                     downsampledChunksToPersist, userTimeStart, userTimeEnd)
+    try {
+      rawPartsBatch.foreach { rawPart =>
+        downsamplePart(rawPart, rawPartsToFree, downsampledPartsPartsToFree,
+          downsampledChunksToPersist, userTimeStart, userTimeEnd)
+      }
+      persistDownsampledChunks(downsampledChunksToPersist)
+    } finally {
+      // reclaim all blocks
+      offHeapMem.get().blockMemFactory.markUsedBlocksReclaimable()
```

Comment:
Probably should move this to a close/free method on the offHeapMemObject.

Reply:
Right now, I don't think any abstraction is offered by the offHeapMemObject. It is just a place holder for thread locals. The user of the block mem is responsible for releasing.